addons/portal/controllers/thread.py

```diff
@@ -15,3 +15,12 @@ def _prepare_post_data(self, post_data, thread, **kwargs):
             ):
                 post_data["author_id"] = partner.id
         return post_data
+
+    def _is_message_editable(self, message, guest, **kwargs):
+        thread = request.env[message.model].browse(message.res_id)
+        if thread and request.env.user._is_public():
```

Comment:
Note (we can tweak it later):

Enforcing "is public" is a bit strange/useless as it's always possible for the current user to use the same URL (with the token) while being logged out to by-pass it.